quantecon/game_theory/repeated_game.py

```diff
@@ -0,0 +1,454 @@
+"""
+Tools for repeated game.
+
+"""
+
+import numpy as np
+from scipy.spatial import ConvexHull
+from numba import jit, njit
+
+
+class RepeatedGame:
+    """
+    Class representing an N-player repeated game.
+
+    Parameters
+    ----------
+    stage_game : NormalFormGame
+                 The stage game used to create the repeated game.
+
+    delta : scalar(float)
+            The common discount rate at which all players discount the future.
+
+    """
+    def __init__(self, stage_game, delta):
+        self.sg = stage_game
+        self.delta = delta
+        self.N = stage_game.N
+        self.nums_actions = stage_game.nums_actions
+
+    def equilibrium_payoffs(self, method=None, options=None):
+        """
+        Compute the set of payoff pairs of all pure-strategy subgame-perfect
+        equilibria with public randomization for any repeated two-player games
+        with perfect monitoring and discounting.
+
+        Parameters
+        ----------
+        method : str, optional
+            The method for solving the equilibrium payoff set.
+
+        options : dict, optional
+            A dictionary of method options. For example, 'abreu_sannikov'
+            method accepts the following options:
+
+                tol : scalar(float)
+                    Tolerance for convergence checking.
+                max_iter : scalar(int)
+                    Maximum number of iterations.
+                u_init : ndarray(float, ndim=1)
+                    The initial guess of threat points.
+
+        Notes
+        -----
+        Here lists all the implemented methods. The default method
+        is 'abreu_sannikov'.
+
+            1. 'abreu_sannikov'
+        """
+        if method is None:
+            method = 'abreu_sannikov'
+
+        if options is None:
+            options = {}
+
+        if method in ('abreu_sannikov', 'AS'):
+            return _equilibrium_payoffs_abreu_sannikov(self, **options)
+        else:
+            msg = f"method {method} not supported."
+            raise NotImplementedError(msg)
+
+
+def _equilibrium_payoffs_abreu_sannikov(rpg, tol=1e-12, max_iter=500,
```

Comment:
Does it make sense to export this function, under the name, say, `abreu_sannikov` (this is what I meant in my previous comment)? In the current version, no documentation is exposed for the AS algorithm implementation.

Reply:
Would it be a little bit redundant if we both export `abreu_sannikov` and `RepeatedGame.equilibrium_payoffs(method='abreu_sannikov')`?

What `scipy.optimize.minimize` does is to make documentation page for each internal function corresponding to each method (e.g. [minimize(method='COBYLA')](https://github.com/scipy/scipy/blob/8dba340293fe20e62e173bdf2c10ae208286692f/doc/source/optimize.minimize-cobyla.rst)) and then add a reference link in the documentation of `scipy.optimize.minimize` (e.g. [here](https://github.com/scipy/scipy/blob/8dba340293fe20e62e173bdf2c10ae208286692f/scipy/optimize/_minimize.py#L74)). Do you think we can follow this way?